### PR TITLE
#475 Minimesos images use Mesos version and minimesos Docker version as tag

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

File: minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java [hunk 1]
```diff
@@ -333,7 +335,7 @@ public void setMapPortsToHost(boolean mapPortsToHost) {
     }
 
     public void waitForState(final Predicate<State> predicate) {
-        Awaitility.await().atMost(clusterConfig.getTimeout(), TimeUnit.SECONDS).until(() -> {
+        await("Predicate on Mesos Master failed").atMost(clusterConfig.getTimeout(), TimeUnit.SECONDS).until(() -> {
             try {
```
Comment (adam-sandor) [inline minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java]: I found a better way of doing these non-trivial waits. You use the version of the callback that doesn't return anything (Runnable) and throw AssertionErrors from it. That way when the timeout expires Awaitility will take the message from the AssertionError showing what exactly has failed.

**Comment by adam-sandor** [comment]
LGTM

File: minimesos/src/main/java/com/containersol/minimesos/mesos/MesosClusterContainersFactory.java [hunk 2]
```diff
@@ -127,6 +133,14 @@ public void destroyRunningCluster(String clusterId) {
         DockerContainersUtil.getContainers(true).filterByName(ContainerName.getContainerNamePattern(clusterId)).kill(true).remove();
     }
 
+    public MesosCluster createMesosCluster(String path) {
+        try (InputStream is = new FileInputStream(path)) {
+            return createMesosCluster(is);
+        } catch (IOException e) {
```
Comment (containersol) [inline minimesos/src/main/java/com/containersol/minimesos/mesos/MesosClusterContainersFactory.java]: CRITICAL Either log or rethrow this exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

**Comment by containersol** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 1 critical
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this unused import 'com.jayway.awaitility.Awaitility'.](https://github.com/ContainerSolutions/minimesos/blob/ad1718008a72b85ca9558e9769c0039ce844781e/minimesos/src/main/java/com/containersol/minimesos/cluster/MesosCluster.java#L24) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Pull request merged**